internal/encoding/decode.go

```diff
@@ -264,7 +268,11 @@ func UnmarshalComposite(r *buffer.Buffer, type_ AMQPType, fields ...UnmarshalFie
 		}
 
 		// Unmarshal each of the received fields.
-		err = Unmarshal(r, field.Field)
+		if field.HandleUnmarshal != nil {
```

Comment:
The types should implement [`unmarshaller`](https://github.com/Azure/go-amqp/blob/main/internal/encoding/decode.go#L18-L20) so we don't need to do this.

Reply:
In this case the problem is that I'm unmarshalling the Target, which is of type 'any'. So this is a way to associate an unmarshaller with a field when the type is not known.

Reply:
Yeah and I think that we should fix that too. Probably similar to how we have the `encoding.DeliveryState` interface for the various disposition frames.

Reply:
```go
// TargetType encapsulates the various target types.
type TargetType interface {
	targetType() // marker method
}
```